src/nbody/main.cpp

```diff
@@ -0,0 +1,587 @@
+    Corrade::PluginManager::Manager<Magnum::Trade::AbstractImporter> manager;
+    auto importer{manager.loadAndInstantiate("AnyImageImporter")};
+    if (importer==nullptr || !importer->openData(rs.getRaw("star.png"))) {
+      throw std::runtime_error{"Failed to load star.png"};
+    }
+    auto image{importer->image2D(0)};
+
+    // Enable setting point size in shader
+    glEnable(GL_PROGRAM_POINT_SIZE);
+
+    // Create star texture and bind it to GL slot 0
+    m_star_tex.setWrapping(Magnum::GL::SamplerWrapping::ClampToEdge)
+              .setMagnificationFilter(Magnum::GL::SamplerFilter::Linear)
+              .setMinificationFilter(Magnum::GL::SamplerFilter::Linear)
+              .setStorage(1, Magnum::GL::textureFormat(image->format()), image->size())
+              .setSubImage(0, {}, *image);
+
+    m_ui_force_coulomb_file.fill(0);
+
+    // Start with a perspective camera looking at the center
+    using namespace Magnum::Math::Literals;
+    m_view = Magnum::Matrix4::lookAt(
+      Magnum::Vector3{200}, Magnum::Vector3{0}, Magnum::Vector3{0,0,1})
+      .invertedRigid();
+    m_viewProjection = Magnum::Matrix4::perspectiveProjection(
+        60.0_degf, Magnum::Vector2{windowSize()}.aspectRatio(), 0.01f, 50000.0f);
+
+    init_gl_bufs();
+
+    // Start the simulation with the application
+    m_ui_initialize = true;
+    m_ui_paused = false;
+  }
+
+  // Initializes the GL buffer for star position data with the current number of
+  // bodies
+  void init_gl_bufs() {
+    const size_t arraySize{m_n_bodies * sizeof(sycl::vec<num_t, 3>)};
+    m_vboStorage = Corrade::Containers::Array<char>(Corrade::ValueInit, 2*arraySize);
+    m_mesh = Magnum::GL::Mesh{Magnum::GL::MeshPrimitive::Points};
+    m_mesh.setCount(m_n_bodies);
+  }
+
+  void tickEvent() override {
+    // Initialize simulation if requested in UI
+    if (m_ui_initialize) {
+      m_n_bodies = m_ui_n_bodies;
+
+      if (m_ui_force_id == UI_FORCE_COULOMB) {
+        printf("Loading Coulomb data from %s\n",
+               m_ui_force_coulomb_file.data());
+
+        // First line is the particle count
+        std::ifstream fin{m_ui_force_coulomb_file.data()};
+        fin >> m_n_bodies;
+
+        // Following lines are particle data
+        std::vector<particle_data<num_t>> particles(m_n_bodies);
+        for (auto& particle : particles) {
+          num_t pos[3];
+          fin >> particle.charge >> pos[0] >> pos[1] >> pos[2];
+          particle.pos = {pos[0], pos[1], pos[2]};
+        }
+
+        m_sim = GravSim<num_t>(m_n_bodies, std::move(particles));
+      } else if (m_ui_distrib_id == UI_DISTRIB_CYLINDER) {
+        m_sim = GravSim<num_t>(
+            m_n_bodies,
+            distrib_cylinder<num_t>{
+                {m_ui_distrib_cylinder_params.min_radius,
+                 m_ui_distrib_cylinder_params.max_radius},
+                {m_ui_distrib_cylinder_params.min_angle_pis * M_PI,
+                 m_ui_distrib_cylinder_params.max_angle_pis * M_PI},
```

Comment:
I think `M_PI` is a POSIX thing. If it's portable fair enough but otherwise it might be better to keep the PI as-is, or calculate it as a `constexpr` variable or something.

Reply:
You're right, that's a good point. I dislike redefining mathematical and physical constants over and over (redundant, messy and bug-prone) but reverted to the original solution for this reason. I'm hoping we'll adapt the demos to C++20 soon and replace it with `std::numbers::pi_v<num_t>`.